pkg/provider-local/controller/worker/machines.go

```diff
@@ -128,6 +129,25 @@ func (w *workerDelegate) generateMachineConfig() error {
 			ProviderSpec: runtime.RawExtension{Raw: []byte(`{"image":"` + image + `"}`)},
 		})
 
+		mcdAnnotationMap := map[string]string{}
+		if pool.Autoscaler != nil {
+			if pool.Autoscaler.ScaleDownUtilizationThreshold != nil {
+				mcdAnnotationMap[ca.ScaleDownUtilizationThresholdAnnotation] = *pool.Autoscaler.ScaleDownUtilizationThreshold
+			}
+			if pool.Autoscaler.ScaleDownGpuUtilizationThreshold != nil {
+				mcdAnnotationMap[ca.ScaleDownGpuUtilizationThresholdAnnotation] = *pool.Autoscaler.ScaleDownGpuUtilizationThreshold
+			}
+			if pool.Autoscaler.ScaleDownUnneededTime != nil {
+				mcdAnnotationMap[ca.ScaleDownUnneededTimeAnnotation] = pool.Autoscaler.ScaleDownUnneededTime.Duration.String()
+			}
+			if pool.Autoscaler.ScaleDownUnreadyTime != nil {
+				mcdAnnotationMap[ca.ScaleDownUnreadyTimeAnnotation] = pool.Autoscaler.ScaleDownUnreadyTime.Duration.String()
+			}
+			if pool.Autoscaler.MaxNodeProvisionTime != nil {
+				mcdAnnotationMap[ca.MaxNodeProvisionTimeAnnotation] = pool.Autoscaler.MaxNodeProvisionTime.Duration.String()
+			}
+		}
```

Comment:
Can you extract this into a helper function in `https://github.com/gardener/gardener/tree/master/pkg/apis/extensions/v1alpha1/helper` that all provider extensions can call to generate this annotation map?